app/models/in_person_enrollment.rb

```diff
@@ -60,17 +60,25 @@ def self.needs_late_email_reminder(early_benchmark, late_benchmark)
   def self.needs_usps_status_check(check_interval)
     where(status: :pending).
       and(
-        where(status_check_attempted_at: check_interval).
-        or(where(status_check_attempted_at: nil)),
+        where(last_batch_claim_at: check_interval).
+        or(where(last_batch_claim_at: nil)),
```

Comment:
When this is first deployed, the job will reprocess all pending enrollment records. We should consider either updating `last_batch_claim_at` in the migration or making this query backwards-compatible by retaining the original filter in addition to the new one.

Be sure to keep `needs_usps_status_check?` up-to-date with changes to this function.

Reply:
I am a fan of the migration idea, setting last_batch_claim_at to the enrollment's previous status_check_attempted_at value.

Reply:
Looks like I can't add this without a hacky way - Postgres doesn't allow using other fields to set defaults.

Reply:
So what's the expected outcome of this first run, where it will re-run all pending enrollments?

Reply:
I think a solution like [this](https://stackoverflow.com/a/5225626) would do nicely. We could limit to to just pending enrollments to make it somewhat faster

Reply:
I would be inclined to make the query backwards-compatible, but I think this is a scenario where it's also alright to run an update query like Sheldon mentioned since we only need to set the new field.